cmake/ChooseBlas.cmake

```diff
@@ -33,7 +34,14 @@ if(USE_BLAS MATCHES "MKL" OR USE_BLAS MATCHES "mkl" OR NOT DEFINED USE_BLAS)
   endif()
 endif()
 
-if(BLAS STREQUAL "Atlas" OR BLAS STREQUAL "atlas")
+if(BLAS STREQUAL "FlexiBLAS" OR BLAS STREQUAL "flexiblas")
+  find_package(BLAS REQUIRED)
```

Comment:
For the other BLAS besides MKL this ChooseBlas.cmake does not rely on the [upstream find_package(BLAS)](https://cmake.org/cmake/help/latest/module/FindBLAS.html). My only concern is that with your PR, users can specify BLAS=FlexiBLAS, but your find_package(BLAS) will find any BLAS that may or may not be FlexiBLAS. That's why it will be helpful for you to force BLA_VENDOR=FlexiBLAS inside this if clause.

So, specifically I'm suggesting adding a `set(BLA_VENDOR FlexiBLAS)` before the current line but still inside the if clause.

Reply:
Right! Just updated and bumped minimum cmake version to support BLA_VENDOR=Flexiblas.

Reply:
Thank you! How about keeping the CMakeLists.txt cmake_minimum_required as is and adding `cmake_minimum_required(VERSION 3.19)` before `set(BLA_VENDOR FlexiBLAS)` inside the if clause? That would avoid unnecessarily breaking users on Ubuntu 20.04 LTS

Reply:
I think it is best to raise the minimum cmake version especially since there's a lot of nice to have and improvement in recent version. My opinion is : _live in the present and look in front, not behind_ in terms of versions.

How would that break for users? (I mean except having to update their cmake version.) 

Have two requirements in different places could be surprising.